smart_open/s3.py

```diff
@@ -553,6 +561,13 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         else:
             self.close()
 
+    def __str__(self):
+        return "s3.BufferedOutputBase(bucket_name='{}', key='{}')".format(
```

Comment:
str should prioritize brevity:

```suggestion
        return "smart_open.s3.BufferedOutputBase(%r, %r)" % (self._object.bucket_name, self._object.key)
```

Also please avoid using the .format() stuff (we don't like it).